src/roles/vsd-deploy/tasks/renew_ejabberd_license.yml

```diff
@@ -10,59 +9,75 @@
-      chown ejabberd:hadoopusers ejabberd_cluster.beam ejabberd_c2s.beam ejabberd.beam ejabberd_sm.beam ejabberd_router.beam ejabberd_listener.beam
-    args:
-      chdir: "{{ ejabberd_base_dir }}/ebin"
-    when: ejabberd_base_dir.stdout is search("3.2.13_5")
+  - name: Run when ejabberd license file is defined
+    block:
+
+    - name: Find the ejabberd base directory
+      command: find /opt/ejabberd/lib/ -type d -name "ejabberd_nuage*"
+      register: ejabberd_base_dir
+
+    - name: Backup the current license files for ejabberd
+      # noqa 303
+      command: >-
+        tar zcf orig_backup.tgz ejabberd.beam ejabberd_admin.beam ejabberd_c2s.beam
+        ejabberd_cluster.beam ejabberd_config.beam ejabberd_listener.beam ejabberd_sm.beam
+      args:
+        chdir: "{{ ejabberd_base_dir.stdout }}/ebin"
+
+    - name: Copy new ejabberd license file to root folder on VSD
+      copy:
+        dest: "/root/"
+        src: "{{ vsd_ejabberd_license_file }}"
+
+    - name: Untar the new ejabberd license file
+      # noqa 303
+      command: "tar zxf /root/{{ vsd_ejabberd_license_file | basename }}"
+      args:
+        chdir: "{{ ejabberd_base_dir.stdout }}/ebin"
+
+    - name: Change ownership of licenses
+      # noqa 302
+      command: >-
+        chown ejabberd:hadoopusers ejabberd_license.beam ejabberd_cluster.beam
+        ejabberd_c2s.beam ejabberd.beam ejabberd_sm.beam ejabberd_router.beam ejabberd_listener.beam
+      args:
+        chdir: "{{ ejabberd_base_dir.stdout }}/ebin"
+      when: not(ejabberd_base_dir.stdout is search("3.2.13_5"))
+
+    - name: Change ownership of licenses for version 3.2.13_5
+      # noqa 302
+      command: >-
+        chown ejabberd:hadoopusers ejabberd_cluster.beam ejabberd_c2s.beam ejabberd.beam ejabberd_sm.beam ejabberd_router.beam ejabberd_listener.beam
+      args:
+        chdir: "{{ ejabberd_base_dir }}/ebin"
+      when: ejabberd_base_dir.stdout is search("3.2.13_5")
+
+    when: vsd_ejabberd_license_file is defined
+
+  - name: Changing log_rotate_size to non-zero
+    lineinfile:
+      path: /opt/ejabberd/conf/ejabberd.yml
+      regexp: '^log_rotate_size:'
+      line: "log_rotate_size: 10485760"
+    when: "vsd_version.stdout is version('20.10.1', '>=')"
 
   - name: Start ejabberd
     command: /opt/ejabberd/bin/ejabberdctl start
-    register: license_check_after_renew
 
   - name: Check that new ejabberd license is valid
     command: /opt/ejabberd/bin/ejabberdctl license_info
     register: license_check_after_renew
-    retries: 5
+    retries: 20
     until: "license_check_after_renew.rc == 0"
     delay: 20
+    ignore_errors: yes
 
   - name: Stop ejabberd service
     command: /opt/ejabberd/bin/ejabberdctl stop
 
-  when: vsd_ejabberd_license_file is defined
+  - name: Check if the license has expired
+    assert:
+      that: "license_check_after_renew.rc == 0"
+      msg: "Problem checking the expiration date of the ejabberd license. Please check if it has expired."
```

Comment:
I think it would be good to provide some more information here - maybe add the output in license_check_after_renew?

Reply:
I made that change, please review it again.